src/coprocessor/dag/batch/executors/util/aggr_executor.rs

```diff
@@ -304,6 +307,39 @@ impl<Src: BatchExecutor, I: AggregationExecutorImpl<Src>> BatchExecutor
     }
 }
 
+/// Decodes all columns that are not decoded.
+pub fn ensure_columns_decoded(
+    tz: &Tz,
+    exprs: &[RpnExpression],
+    schema: &[FieldType],
+    input: &mut LazyBatchColumnVec,
+) -> Result<()> {
+    for expr in exprs {
+        expr.ensure_columns_decoded(tz, schema, input)?;
+    }
+    Ok(())
+}
+
+/// Evaluates expressions and outputs the result into the given Vec. Lifetime of the expressions
+/// are erased.
+pub unsafe fn eval_exprs_no_lifetime<'a>(
+    context: &mut EvalContext,
+    exprs: &[RpnExpression],
+    schema: &[FieldType],
+    input: &LazyBatchColumnVec,
+    output: &mut Vec<RpnStackNode<'a>>,
+) -> Result<()> {
+    for expr in exprs {
+        output.push(erase_lifetime(expr).eval_unchecked(
```

Comment:
It's time to rename `eval_unchecked` to `eval_decoded`. 🤔

Reply:
Feel good. Changed.